### DISCUSSION TITLE
All kinds of new errors started today

### DISCUSSION THREAD

**Discussion by mighty-hq** [directus/directus#17461] [Archive]
### Describe the Bug

Hello, 2 big errors started happening today:

1. When loading a flow, the logs do not load on the right menu intermittently. After waiting about 30 seconds, they either do load, or I get an error "[UNKNOWN] Request failed with status code 504"

2.When creating a new field in a data model, if I create a simple input field, it works. However, if I create a many to one field, after selecting the collection, I get error " [INVALID_PAYLOAD] Field "store" doesn't exist in the collection "activity""
Keep in mind, that is the field I am trying to create. Once I close error and refresh page, that field does show up as created, but it is a normal string field. 

### To Reproduce

For problem #1
-Click Settings
-Click Flows
-Wait

For problem #2
-Click data model
-Click "Create field in Advanced Mode"
-Select Many To One Relationship
-Type Unique column name
-Click "Relationship" in left menu
-Select related collection
-Confirm

### Errors Shown

1. "[UNKNOWN] Request failed with status code 504"
2." [INVALID_PAYLOAD] Field "store" doesn't exist in the collection "activity""

### What version of Directus are you using?

9.22.1

### What version of Node.js are you using?

18

### What database are you using?

SQL

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by mighty-hq** [comment]
Did a test where I create the many to one field and also check the "Add O2M to "Store", and I get no error and it is added succesfully

**Comment by mighty-hq** [comment]
Another strange thing I just noticed. I added a new field to a collection. When querying with the api, I get the following error "Cannot query field \"internal_note\" on type \"Ticket\"". However, that is also intermittently.

**Comment by paescuj** [comment]
Is there a proxy placed in between? What database do you mean with "SQL"? Any chance to update to the latest version (currently v9.22.4)?

**Reply by mighty-hq** [reply]
Sorry, what I meant is the database is mysql. (DB_CLIENT="mysql").  There is no proxy. Also, we have not made any changes or updates on our end that could of tweaked something, these issues just started in the last 24 hours.

**Reply by mighty-hq** [reply]
Just curious since this is my first issue submitted, why was the ticket closed and moved to discussion, if its still a problem?

**Reply by paescuj** [reply]
It doesn't sound like a bug on the part of Directus itself. Therefore the issue has been transformed into a discussion to find out whether there is possibly another problem. Should it turn out that it's actually a bug within Directus, this will be converted back into an issue.

**Reply by paescuj** [reply]
> There is no proxy

Also no webserver in front of it?

**Reply by mighty-hq** [reply]
Got it. Ok so I just updated to version 9.22.3, and same issue.  The current setup is Docker deployed to AWS ECS ran on ECS Fargate. The webserver is also on AWS.  Does that answer your question? If you need other details please let me know, thank you. 

What I find weird is that after adding a new field, when I do graphql api calls, sometimes it fails, sometimes it doesnt.  That's the (Cannot query field "internal_note" on type "Ticket"") error.

**Reply by paescuj** [reply]
> Got it. Ok so I just updated to version 9.22.3, and same issue. The current setup is Docker deployed to AWS ECS ran on ECS Fargate. The webserver is also on AWS. Does that answer your question? If you need other details please let me know, thank you.

Okay, thanks for the info! This is what I basically meant by a proxy / webserver. The [504 error (Gateway Timeout)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504) you're receiving sometimes indicates there's a problem with the set-up of the proxy and not Directus itself.

**Reply by mighty-hq** [reply]
Its so strange, because we have not updated anything. By any chance, does the following give you a clearer hint on what could be happening?

When I create a new test field in the data model (many to one) without O2M on the related collection, again, I get the error, that the field I am trying to create does not exist, however, it does get created and the orange alert triangle shows up on the field "Database only, click to configure". 

![image](https://user-images.githubusercontent.com/81202462/217799819-0e32afb5-a13a-482b-8fdb-7166595c6b26.png)

**Reply by mighty-hq** [reply]
Oh, I believe I found the issue. So, as mentioned before, the hosting is on AWS fargate, where you mention how many tasks you want to run concurrently. I had 2 , and I believe what was happening, is it was doing the api calls on the separate tasks. 
Create M2O action went to Task #1.
Create O2M action went to Task #2. 

But because they went at the same time, the Initial field was not yet created on Task#1 for it to be referenced. 

I hope what I am trying to explain makes sense. If you have any advice on how I can fix that, that would be great. If not, I will reach out to AWS, I appreciate you.

**Reply by mighty-hq** [reply]
Forgot to mention, I manually updated the instance to only run 1 task, and the error dissapeared when creating M2O fields.

**Reply by paescuj** [reply]
This actually sounds like it could be the cause of the problem. Unfortunately I cannot really help you in this matter, but perhaps someone from the community or AWS support as you wrote.
